stream-legal-entity/legal-entity-core/src/main/java/com/backbase/stream/LegalEntitySaga.java

```diff
@@ -260,21 +263,29 @@ private ProductGroupTask createProductGroupTask(LegalEntityTask streamTask, Prod
         return new ProductGroupTask(streamTask.getId() + "-" + productGroup.getName(), productGroup);
     }
 
-    @ContinueSpan(log = "createReferenceJobRoles")
-    private Mono<LegalEntityTask> createReferenceJobRoles(LegalEntityTask streamTask) {
-        if (isEmpty(streamTask.getData().getReferenceJobRoles())) {
-            log.debug("Skipping creation of reference job roles.");
+    @ContinueSpan(log = "createJobRoles")
+    private Mono<LegalEntityTask> createJobRoles(LegalEntityTask streamTask) {
+        if (isEmpty(streamTask.getData().getReferenceJobRoles())
+            && isEmpty(streamTask.getData().getMasterServiceAgreement().getJobRoles())) {
+            log.debug("Skipping creation of job roles.");
             return Mono.just(streamTask);
         }
 
-        log.info("Creating Reference Job Roles...");
+        log.info("Creating Job Roles...");
         LegalEntity legalEntity = streamTask.getData();
-        ServiceAgreement masterServiceAgreement = legalEntity.getMasterServiceAgreement();
-        return Flux.fromIterable(legalEntity.getReferenceJobRoles())
-            .flatMap(referenceJobRole ->
-                accessGroupService.setupReferenceJobRole(streamTask, masterServiceAgreement, referenceJobRole))
-            .flatMap(referenceJobRole -> {
-                log.debug("Reference Job Role: {}", referenceJobRole.getName());
+        ServiceAgreement serviceAgreement = legalEntity.getMasterServiceAgreement();
+
+        Stream<JobRole> jobRoles = serviceAgreement.getJobRoles() == null
```

Comment:
Stream.of(serviceAgreement.getJobRoles(), legalEntity.getReferenceJobRoles()).filter(Objects.nonNull).flatMap(Collection::stream)

Reply:
done!